### PR TITLE
Support for proxies and CLI applications

### PR DESCRIPTION
**Pull request by jlubeck**
Basically what the title says. Simple changes, which expand the possibilities of the package.

### CODE REVIEW

File: lib/models/request_data.dart [hunk 1]
```diff
@@ -14,8 +13,8 @@ class RequestData {
   Encoding encoding;
 
   RequestData({
-    @required this.method,
-    @required this.baseUrl,
```
Comment (CodingAleCR) [inline lib/models/request_data.dart:18]: Nice job, just a question, why did you remove the @required decorator here?

**Comment by jlubeck** [comment]
It looks like that decorator is coming from flutter/foundation. And that import is not possible to use on CLI applications. And since that is that just a decorator (you still had the asserts there) I see no harm in removing that if it allows support for CLI apps. Hopefully you agree!

**Comment by CodingAleCR** [comment]
Yes, it definitely should not be flutter foundation import. I probably missed it when I changed the different @required decorators to be from the meta plugin. This plugin allows you to work with CLI applications. I'll add it it the future since as you said there's not much of a difference in terms of functionality, it's more of a semantics issue. 

Thanks and great work!

**Review by CodingAleCR — APPROVED** [review]
